File: Algorithms/Game Theory/Nimble Game.php
<?php

/*
 * Complete the 'nimbleGame' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER_ARRAY s as parameter.
 */

function nimbleGame($s) {
    # Write your code here
    $xor_sum = 0;
    foreach ($s as $i => $coins) {
        if ($coins % 2 != 0) {
            $xor_sum ^= $i;
        }
    }

    return ($xor_sum == 0) ? "Second" : "First";
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$t = intval(trim(fgets(STDIN)));

for ($t_itr = 0; $t_itr < $t; $t_itr++) {
    $n = intval(trim(fgets(STDIN)));

    $s_temp = rtrim(fgets(STDIN));

    $s = array_map('intval', preg_split('/ /', $s_temp, -1, PREG_SPLIT_NO_EMPTY));

    $result = nimbleGame($s);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);

File: Algorithms/Game Theory/Nimble Game.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'nimbleGame' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts INTEGER_ARRAY s as parameter.
     */

    public static string nimbleGame(List<int> s)
    {
        int xorSum = 0;
        for (int i = 0; i < s.Count; i++)
        {
            if (s[i] % 2 != 0)
            {
                xorSum ^= i;
            }
        }

        return xorSum == 0 ? "Second" : "First";
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int t = Convert.ToInt32(Console.ReadLine().Trim());

        for (int tItr = 0; tItr < t; tItr++)
        {
            int n = Convert.ToInt32(Console.ReadLine().Trim());

            List<int> s = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(sTemp => Convert.ToInt32(sTemp)).ToList();

            string result = Result.nimbleGame(s);

            textWriter.WriteLine(result);
        }

        textWriter.Flush();
        textWriter.Close();
    }
}
